lib/slither/generator.rb

```diff
@@ -20,6 +20,20 @@ def generate(data)
       generate_file_output
     end
 
+    def stream(data, io)
+      first_line = true
+      @definition.sections.each do |section|
+        content = data[section.name]
+        raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") if !content&.any? && !section.optional
+
+        content = [content] if content.is_a?(Hash)
+        content&.each do |row|
+          io << (section.format(row).prepend(new_line_needed(first_line))).freeze
+          first_line = false
+        end
+      end
```

Comment:
Since this part is more or less is the same as the body of https://github.com/Kaligo/slither/blob/cd531e2c6c2c9459edecdecc6af199a8a98eff15/lib/slither/generator.rb#L9 maybe we could merge them?

Reply:
I would not mix the two. For me these are different approaches.